src/storage/fileio/unix/mod.rs

```diff
@@ -0,0 +1,195 @@
+use nix::libc;
+use std::fs::OpenOptions;
+use std::os::unix::prelude::FileExt;
+use std::{fs::File, io};
+
+use super::FileIO;
+
+pub struct UnixFileIO {}
+
+impl UnixFileIO {
+    #[cfg(target_os = "macos")]
+    fn open_or_create_file(path: String, write: bool, create: bool) -> io::Result<File> {
```

Comment:
This only works on MacOS and not standard Linux right? If so, I think `MacFileIO` should be a separate unit from `LinuxFileIO`.

Reply:
If this is needed in the future, we can add it - at this point, not worth that level of granularity for one function (e.g where do we stop? pointer width, different linux distributions etc.)

Reply:
This is similar to how rust std handles multiplatform as well